src/main/java/hudson/remoting/UserRequest.java

```diff
@@ -90,6 +90,17 @@ public UserRequest(Channel local, Callable<?,EXC> c) throws IOException {
         this.classLoaderProxy = RemoteClassLoader.export(cl, local);
     }
 
+    @Override
+    public void checkIfCanBeExecutedOnChannel(Channel channel) throws IOException {
+        // Default check for all requests
+        super.checkIfCanBeExecutedOnChannel(channel);
+        
+        // We also do not want to run UserRequests when the channel is being closed
+        if (channel.isClosingOrClosed()) {
+            throw new IOException("The request cannot be executed on channel " + channel + ". The channel is closing down or has closed down");
```

Comment:
@jglick Yes and No. `Channel#call()` is a wrapper method for calling `Callable`s by wrapping them into `UserRequest`s. All other Request types and user-created requests can be executed only via the low level API. So I can imagine use-cases when somebody needs to call them directly. from outside Remoting 

Withing Remoting these methods are being used to invoke RPCRequests from RemoteInvocationHandler. Ans such calls should be definitely protected as well.